src/main/scala/cromwell/engine/WorkflowDescriptor.scala

```diff
@@ -82,79 +91,116 @@ case class WorkflowDescriptor(id: WorkflowId,
   lazy val writeToCache = configCallCaching && optionCacheWriting
   lazy val readFromCache = configCallCaching && optionCacheReading
 
-  lazy val workflowLogger = props.get("LOG_MODE") match {
-    case Some(x) if x.toUpperCase.contains("SERVER") => makeFileLogger(
-      Paths.get(props.getOrElse("LOG_ROOT", ".")),
-      s"workflow.$id.log",
-      Level.toLevel(props.getOrElse("LOG_LEVEL", "debug"))
-    )
-    case _ => NOPLogger.NOP_LOGGER
+  lazy val workflowLogName = s"workflow.$id.log"
+  lazy val workflowLogPath = props.get("LOG_MODE") match {
+    case Some(x) if x.toUpperCase.contains("SERVER") =>
+      Option(File(props.getOrElse("LOG_ROOT", ".")) / workflowLogName) map (_.path)
+    case _ => None
+  }
+  lazy val workflowLogger = workflowLogPath match {
+    case Some(path) => makeFileLogger(path, workflowLogName, Level.toLevel(props.getOrElse("LOG_LEVEL", "debug")))
+    case None => NOPLogger.NOP_LOGGER
   }
 
   lazy val workflowRootPath = wfContext.root.toPath(gcsFilesystem)
-  def workflowRootPathWithBaseRoot(rootPath: String): Path =
+  def workflowRootPathWithBaseRoot(rootPath: String): Path = {
     WorkflowDescriptor.buildWorkflowRootPath(rootPath, name, id).toPath(gcsFilesystem)
+  }
 
-  def copyWorkflowOutputs(implicit executionContext: ExecutionContext): Future[Unit] = {
+  def copyWorkflowOutputs(workflowMetadataResponse: WorkflowMetadataResponse)
```

Comment:
This one I have more trouble with although I'm also having difficulty staying self-consistent between my idealist and realist hats. I'm getting a little leery of continuing to throw any old thing into workflowdescriptor on the hope that it gets cleaned up as part of the other sweeping changes.

OTOH over the last couple of days it is looking increasingly likely that the plan will be "make it work, now fix it" in regards to the backend changes so it is more likely that cleaning this up will actually happen